OpenEMS/openems
# FEMS Backport

- KACO PV-Inverter: calculate production energy manually
- GoodWe 20/30 Chargers: avoid multiple HIGH Priority mobus tasks
- UI: Utils, tests and translations
-  UI: Fix chart wrong scaling when firstSetupProtocol is too close to...
-  Backport FEMS Backend
    - Fix compatibility with and require Odoo 16 (separate Odoo module will be updated independently)
    - Fix possible NPEs
    - Fix starting of Websocket servers (avoid event race conditions)
    - Add COMPONENT_IDs to backend components
    - Fix Websocket handshake case-insensitive
    - Improve general performance
    - Avoid excessive logs
    - Add GenericSystemLog to Metadata: record system execute and update
    - Improve InfluxDB Aggregated data handling
    - Allow multiple InfluxDB servers for different periods (defined by start-/enddate)
    - InfluxDB: only allow Channel-Addresses in standardized format
- Websocket Api Controller: cleanup getEdgeRequest & JUnit tests 
- AppCenter: add check for peakshaving to not be compatible with home 
- AppCenter: separated AppConfiguration into tasks 
- Update Gitpod to Odoo v16 for https://github.com/OpenEMS/odoo-openems/tree/16.0
